olp-cpp-sdk-core/include/olp/core/http/HttpResponse.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <string>
+
+#include <olp/core/CoreApi.h>
+#include <olp/core/http/Network.h>
+#include <olp/core/http/NetworkTypes.h>
+
+namespace olp {
+namespace http {
+
+/**
+ * @brief Response from a REST request.
+ */
+struct CORE_API HttpResponse {
+ public:
+  HttpResponse() = default;
+  HttpResponse(http::ErrorCode status, std::string response = std::string())
+      : status(status), response(std::move(response)) {}
+
+  /**
+   * @brief status Status of the HTTP request.
+   * @see olp::network::Network::ErrorCode for negative status numbers
+   * @see olp::network::HttpStatusCode for positive status numbers
+   */
+  http::ErrorCode status{http::ErrorCode::UNKNOWN_ERROR};
```

Comment:
But then use `int` same as http::NetworkResponse and not ErrorCode which is an enum. This way you limit the scope.